echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/pagerduty/PagerDutyNotificationService.groovy

```diff
@@ -50,17 +58,91 @@ class PagerDutyNotificationService implements NotificationService {
 
   @Override
   void handle(Notification notification) {
+    Map details = new LinkedHashMap()
+    details.put("details", notification.additionalContext.details)
+
+    Map<String, String> errors = new HashMap()
+    Map<String, String> pdErrors = new HashMap()
+
     notification.to.each {
-      pagerDuty.createEvent(
-        "Token token=${token}",
-        new PagerDutyService.PagerDutyCreateEvent(
-          service_key: it,
-          client: "Spinnaker (${notification.source.user})",
-          description: notification.additionalContext.message
+      try {
+        Map response = pagerDuty.createEvent(
+          "Token token=${token}",
+          new PagerDutyService.PagerDutyCreateEvent(
+            service_key: it,
+            client: "Spinnaker (${notification.source.user})",
+            description: notification.additionalContext.message,
+            details: details
+          )
         )
-      )
 
-      log.info('Sent page {} {}', kv('serviceKey', it), kv('message', notification.additionalContext.message))
+        if (response.status.equals("success")) {
```

Comment:
On the off chance there's a null in here ... you could:

`"success".equals(response?.status)`